src/cryptography/hazmat/primitives/interfaces/__init__.py

```diff
@@ -121,70 +123,34 @@ def copy(self):
         """
 
 
-@six.add_metaclass(abc.ABCMeta)
-class RSAPrivateKey(object):
-    @abc.abstractmethod
-    def signer(self, padding, algorithm):
-        """
-        Returns an AsymmetricSignatureContext used for signing data.
-        """
+# DeprecatedIn08
 
-    @abc.abstractmethod
-    def decrypt(self, ciphertext, padding):
-        """
-        Decrypts the provided ciphertext.
-        """
-
-    @abc.abstractproperty
-    def key_size(self):
-        """
-        The bit length of the public modulus.
-        """
-
-    @abc.abstractmethod
-    def public_key(self):
-        """
-        The RSAPublicKey associated with this private key.
-        """
-
-
-@six.add_metaclass(abc.ABCMeta)
-class RSAPrivateKeyWithNumbers(RSAPrivateKey):
-    @abc.abstractmethod
-    def private_numbers(self):
-        """
-        Returns an RSAPrivateNumbers.
-        """
-
-
-@six.add_metaclass(abc.ABCMeta)
-class RSAPublicKey(object):
-    @abc.abstractmethod
-    def verifier(self, signature, padding, algorithm):
-        """
-        Returns an AsymmetricVerificationContext used for verifying signatures.
-        """
-
-    @abc.abstractmethod
-    def encrypt(self, plaintext, padding):
-        """
-        Encrypts the given plaintext.
-        """
-
-    @abc.abstractproperty
-    def key_size(self):
-        """
-        The bit length of the public modulus.
-        """
+RSAPrivateKey = utils.deprecated(
+    rsa.RSAPrivateKey,
+    __name__,
+    "The RSAPrivateKey interface has moved to the cryptography.hazmat.primitives.asymmetric.rsa module",
+    DeprecationWarning
```

Comment:
They should really just use the `DeprecatedIn08` or whatever constants.

On Thu Jan 08 2015 at 3:44:10 AM Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/hazmat/primitives/interfaces/__init__.py
> <https://github.com/pyca/cryptography/pull/1592#discussion-diff-22647306>:
>
> > -    @abc.abstractmethod
> > -    def encrypt(self, plaintext, padding):
> > -        """
> > -        Encrypts the given plaintext.
> > -        """
> > -
> > -    @abc.abstractproperty
> > -    def key_size(self):
> > -        """
> > -        The bit length of the public modulus.
> > -        """
> > +RSAPrivateKey = utils.deprecated(
> > +    rsa.RSAPrivateKey,
> > +    __name__,
> > +    "The RSAPrivateKey interface has moved to the cryptography.hazmat.primitives.asymmetric.rsa module",
> > +    DeprecationWarning
>
> these should be pending deprecation warnings
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1592/files#r22647306>.
>